src/main/kotlin/de/studiocode/miniatureblocks/utils/HashUtils.kt

```diff
@@ -6,15 +6,8 @@ import java.security.MessageDigest
 object HashUtils {
     
     fun createSha1Hash(file: File): ByteArray {
-        val inputStream = file.inputStream()
         val md = MessageDigest.getInstance("SHA1")
-        var len = -1
-        val buffer = ByteArray(4096)
-        while ({len = inputStream.read(buffer); len}() != -1) {
-            md.update(buffer, 0, len)
-        }
-        inputStream.close()
-        return md.digest()
+        return md.digest(file.readBytes())
```

Comment:
This would read the whole file into a byte array which will cause memory problems with big files.

"Gets the entire content of this file as a byte array.
**This method is not recommended on huge files. It has an internal limitation of 2 GB byte array size.**"

Reply:
The engineUpdate method will also save the data in a buffer. So it will still have that 2 GB limit since arrays save the size in a signed 32-bit integer. However that limit shouldn't really matter since resourcepacks can't really reach 2 GB of file size (after compression).

Source of the engineUpdate method can be found [here](https://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/9d617cfd6717/src/share/classes/sun/security/provider/DigestBase.java)

Reply:
Nope, just tried it and the hash function works perfectly with larger files, your version doesn't. (Would be weird if you'd have to load the whole file into ram just to create a hash)